### DISCUSSION TITLE
Implement `bracketSpacing` from Prettier

### DISCUSSION THREAD

**Discussion by isuro** [biomejs/biome#392] [Ideas]
The only missing rule from Prettier that would prevent migrating my ~30k file codebase is [`bracketSpacing`](https://prettier.io/docs/en/options.html#bracket-spacing), without which Biome adds spaces to every object. Any chance of implementing this?

(The only other non-rule functionality we use is per-file rule overrides, i.e. being able to change max line length for specific files, but that's potentially less crucial.)

Thanks!

**Comment by ematipico** [comment]
I want to share Prettier 's vision around options

https://prettier.io/docs/en/option-philosophy.html

**Comment by aweary** [comment]
@ematipico but `bracketSpacing` is one of the option Prettier provides, and not having it does make adopting `biome`  a non-starter for larger codebases that don't want to thrash the commit history with a huge style change. I think Prettier compatibility would go a long way for adoption.

**Reply by Conaclos** [reply]
If there is enough interest for the feature, we will consider it. For now, it seems that the interest of the Biome community is pretty low.

**Comment by ematipico** [comment]
> larger codebases that don't want to thrash the commit history with a huge style change

This isn't necessarily true. With git, you can ignore commits of this kind. We do that in our codebase: https://github.com/biomejs/biome/blob/main/.git-blame-ignore-revs#L8

**Comment by Conaclos** [comment]
This is now [implemented](https://biomejs.dev/reference/configuration/#javascriptformatterbracketspacing).
